sematic/db/queries.py

```diff
@@ -133,6 +188,16 @@ def save_graph(runs: List[Run], artifacts: List[Artifact], edges: List[Edge]):
     """
     Update a graph
     """
+    with db().get_session() as session:
+        for run in runs:
+            existing_run = session.query(Run).filter(Run.id == run.id).one_or_none()
+            if existing_run is not None:
+                if len(run.external_jobs) < len(existing_run.external_jobs):
+                    raise ValueError(
+                        f"Cannot remove existing external jobs from {run.id}. "
+                        f"Existing run had: {existing_run.external_jobs}. New "
+                        f"run had: {run.external_jobs}"
+                    )
```

Comment:
Same here.

And if this is really necessary, DRY with `save_run`.

Reply:
consolidated